python/revng/scripting/project.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import IO, Tuple
+
+import yaml
+
+from revng.model import Binary
+
+
+class Project(ABC):
+    @abstractmethod
+    def fetch(self, *args, **kwargs) -> bytes:
+        pass
+
+    @abstractmethod
+    def fetch_many(self, *args, **kwargs) -> dict:
+        pass
+
+    @abstractmethod
+    def analyze(self, *args, **kwargs) -> Binary:
+        pass
+
+    @abstractmethod
+    def save(self):
+        pass
+
+    def set_model(self, model: Binary):
+        self.model: Binary = model
+        self.last_saved_model: Binary = deepcopy(model)
+
+        for f in self.model.Functions:
+            f.set_project(self)
+        for t in self.model.TypeDefinitions:
+            t.set_project(self)
+        self.model.set_project(self)
+
+    def parse_pipeline_description(self, pipeline_description: IO):
+        self.pipeline_description = yaml.safe_load(pipeline_description)
+
+    def _get_analysis_inputs(self, analysis_name: str) -> Tuple[str | None, str | None]:
+        for step in self.pipeline_description["Steps"]:
+            for analysis in step["Analyses"]:
+                if analysis["Name"] == analysis_name:
+                    if not analysis["ContainerInputs"]:
+                        return None, None
+                    return (
+                        analysis["ContainerInputs"][0]["Name"],
+                        analysis["ContainerInputs"][0]["AcceptableKinds"][0],
```

Comment:
All `ContainerInputs` need to be returned, not just the first one

Reply:
Can there be more that one? I didn't see any of those in the `pipeline-description.yml`. How should the query be constructed in the case of multiple `ContainerInputs`? Should I pass all the inputs to the `$container` dict that I pass to the query? Bellow is an example mutation qurty.

```
mutation($step: String!, $analysis: String!, $container: String!, $index: BigInt!) {
  runAnalysis(step: $step, analysis: $analysis,
    containerToTargets: $container, index: $index) {
      __typename
  }
}
```